src/mgmt/srv.c

```diff
@@ -367,6 +367,32 @@ ds_mgmt_hdlr_svc_rip(crt_rpc_t *rpc)
 	kill(getpid(), sig);
 }
 
+void ds_mgmt_pool_get_svcranks_hdlr(crt_rpc_t *rpc)
+{
+	struct mgmt_pool_get_svcranks_in	*in;
+	struct mgmt_pool_get_svcranks_out	*out;
+	int					 rc;
+
+	in = crt_req_get(rpc);
+	D_ASSERT(in != NULL);
+
+	D_DEBUG(DB_MGMT, "get svcranks for pool "DF_UUIDF"\n",
+		DP_UUID(in->gsr_puuid));
+
+	out = crt_reply_get(rpc);
+
+	rc =  get_pool_svc_ranks(in->gsr_puuid, &out->gsr_ranks);
```

Comment:
Two things:
  1) Not sure if it's worth checking for `out->gsr_ranks == NULL`, given that we should be checking everything in `get_pool_svc_ranks()`, but I thought I'd raise it
  2) I believe that you need to free these ranks after sending the reply

Reply:
I think CaRT may be taking care of the memory management for the d_rank_list_t pointers that are defined in the definition macros of RPCs such as this one. There is a function crt_proc_d_rank_list_t() in crt_hg_proc.c that I think is invoked for encoding, decoding, and freeing (addressing item 2 if I understand right). @frostedcmos can you confirm after the reply is sent the d_rank_list_t * will be freed?

On input to get_pool_svc_ranks() &out->gsr_ranks should be non-NULL. In the event get_pool_svc_ranks() fails, since it doesn't set *svc_ranks, it would be important that out->svc_ranks = NULL (perhaps it is as a typical effect of initializing the crt_rpc_t). On error, if out->gsr_ranks = NULL the crt encoding will handle that, encoding a reply with nr=0. Some tests do exercise errors, such as evict_test.py providing a bogus pool UUID. Let me inspect the RPC replies on those beyond the nonzero rc.

Reply:
I was wrong on the memleak in the handler. Will push another patch freeing the ranks after sending the reply.